src/decimal.jl

```diff
@@ -1,10 +1,27 @@
 # Convert a string to a decimal, e.g. "0.01" -> Decimal(0, 1, -2)
 function Base.parse(::Type{Decimal}, str::AbstractString)
-    if 'e' in str
-        return parse(Decimal, scinote(str))
+    # Read sign
+    s = (str[1] == '-') ? 1 : 0
+    # Unpack scientific notation
+    scno = split(str, 'e')
```

Comment:
IMO longer / more descriptive variable names would be really nice, although I know this repo is bad about this in general (mea culpa...).

Reply:
I tried making the variable names more suggestive. I also added checks for excluding double occurrences of `'.'` or `'e'` in the string.
About the exponent separator for scientific notation, originally it was just `'e'`, but I believe that also `'E'` should be allowed since I've seen many tabulated data using it. I just added a call to `lowercase` at the beginning for this. If you do not agree I can revert it.